src/app/project/comments/comments.component.ts

```diff
@@ -174,20 +193,27 @@ export class CommentsComponent implements OnInit, OnDestroy {
     this.getPaginatedComments(this.tableParams.currentPage);
   }
 
-  public goToItem(item) {
+  /**
+	 * Event handler for a user click on a document
+   * 
+	 * @param item The item that's been clicked on, either a document or external link entry
+	 * @returns {Observable<Object>}
+	 */
+  public goToItem(item: CommentPeriodFile): void {
     // If we're opening an external link instead of a document, go straight to the predefined link
     if (item.externalLink) {
       window.open(item.externalLink);
+    } else {
+      // Otherwise, if it's a regular document, fetch it via the API
+      let filename = item.documentFileName;
```

Comment:
```suggestion
      let safeName = item.documentFileName;
```

Reply:
I can't take credit for this extra line but I should have caught it. Revised as expected.